src/NLog/Layouts/JsonLayout.cs

```diff
@@ -116,10 +120,18 @@ protected override string GetFormattedMessage(LogEventInfo logEvent)
                 }
             }
 
-            AppendIf(!this.SuppressSpaces, sb, " ");
-            sb.Append("}");
+            var result = sb.ToString();
+
+            if (string.IsNullOrEmpty(result.Trim()) && !RenderEmptyObject)
```

Comment:
It's maybe even better to keep a boolean value tracking the "empty state"? 

Less String operations are better (e.g. trim and string comparisons)